### ISSUE TITLE
Support for durable functions

### ISSUE THREAD

**Issue by benbarnett02** [dotnet/aspire#6760]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

Currently, when adding a durable function to an Aspire project, 500 errors are returned when using the built-in (to durable functions) http endpoints to get and modify the function run.

### Describe the solution you'd like

Supporting Durable functions (& bindings etc) would be a great next step now that Aspire supports Azure Functions. My use case for this is a series of long-running functions (5-10 minutes) that perform various data analytics/ETL type actions.

**Comment by joperezr** [comment]
Thank you for logging this @benbarnett02. We should consider this as one of the important things to get now that we are adding Functions support. Adding this to the backlog.

**Comment by mattchenderson** [comment]
It looks like this might be connected to an older version of the Durable extension. Could you try updating the version of `Microsoft.Azure.Functions.Worker.Extensions.DurableTask` to 1.2.2? The default templates are referencing an older version, and we'll update those to use the new version as well.

Another issue we're aware of that you might hit when trying things out is that the deployed app by default would not have sufficient permissions over the storage account (which Durable Functions leverages by default). Aspire does not add any permissions based on the triggers and bindings being used. You would therefore need to add role assignments in your app host's `Program.cs` to enable Durable scenarios. [This documentation section](https://learn.microsoft.com/azure/azure-functions/functions-reference?tabs=blob&pivots=programming-language-csharp#connecting-to-host-storage-with-an-identity) highlights some example built-in roles, as well as general guidance for identity-based connections for Functions host storage. Another option would be to use one of the other state provider options for Durable, but you'd also need to set up those connections.

We'll look at ways to remove that friction point.

**Comment by benbarnett02** [comment]
I've just tested 1.2.2 of `Microsoft.Azure.Functions.Worker.Extensions.DurableTask` and it's working perfectly locally! I haven't tested deployment, but from those docs it looks pretty simple to configure roles on the storage account. 

Not a hard fix at all to bump one package's version, but would be good if there was some clear documentation, maybe [here](https://learn.microsoft.com/en-us/dotnet/aspire/serverless/functions?tabs=dotnet-cli&pivots=visual-studio) to say that durable functions work from 1.2.2 of that package. Happy to raise a PR or issue in those docs if that's the right place for it?

**Comment by maddymontaquila** [comment]
This is a post-GA thing for "officially" supporting it with functions, but we did unblock some functionalilty in 9.2 so it "mostly" works

**Comment by maddymontaquila** [comment]
Durable Functions support has been implemented — `src/Aspire.Hosting.Azure.Functions/DurableTask/` contains `DurableTaskSchedulerResource`, `DurableTaskHubResource`, and supporting types/extensions. Closing as implemented.
